### PR TITLE
notify user if directory contains lots of files on load

### PR DESCRIPTION
**Pull request by kevinushey**
Closes https://github.com/rstudio/renv/issues/1573.

### CODE REVIEW

File: R/snapshot.R [hunk 1]
```diff
@@ -932,15 +941,26 @@ renv_snapshot_dependencies_impl <- function(project, type = NULL, dev = FALSE) {
...
         "Consider using .renvignore to ignore files, or switching to explicit snapshots.",
         "See `?renv::dependencies` for more information.",
+        if (length(count)) c(
+          "",
+          sprintf("- %s: %s", format(names(count)), nplural("file", count))
+        ),
         ""
       )
 
+      renv_scope_options(renv.verbose = TRUE)
```
Comment (hadley) [inline R/snapshot.R:963]: I wonder if this suggests that the `writef()` should actually be a warning?

File: R/snapshot.R [hunk 2]
```diff
@@ -932,15 +941,26 @@ renv_snapshot_dependencies_impl <- function(project, type = NULL, dev = FALSE) {
       if (elapsed < limit)
         return()
 
+      renv_scope_options(renv.verbose = TRUE)
```
Comment (hadley) [inline R/snapshot.R:944]: Duplicated below

File: R/dependencies.R [hunk 3]
```diff
@@ -391,6 +391,9 @@ renv_dependencies_find_dir <- function(path, root, depth) {
   # list children
   children <- renv_dependencies_find_dir_children(path, root, depth)
 
+  # notify about number of children
+  renv_condition_signal("renv.dependencies.count", list(path = path, count = length(children)))
```
Comment (hadley) [inline R/dependencies.R:395]: Is this used anywhere?

File: tests/testthat/test-dependencies.R [hunk 4]
```diff
@@ -462,3 +462,20 @@ test_that("we tolerate calls when parsing dependencies", {
...
+  # create data directory with 'lots' of files
+  dir.create("data")
+  setwd("data")
+  files <- sprintf("%03i.R", 1:200)
+  file.create(files)
+  setwd("..")
+
+  # try to collect snapshot dependencies
+  renv_scope_options(renv.dependencies.elapsed_time_threshold = 0)
+  expect_snapshot(. <- renv_snapshot_dependencies(project))
```
Comment (hadley) [inline tests/testthat/test-dependencies.R:479]: If we stick with the `writef()`, I'd suggest adding a `renv_scope_options(renv.verbose = FALSE)` to assert that the warning is displayed even when not verbose.

**Review by hadley — APPROVED** [review]

File: R/snapshot.R (hunk 1 above)
Comment (kevinushey) [reply R/snapshot.R:963]: I feel like we need something like `ewritef()`, or some other version of `writef()` that ignores `renv.verbose`.

File: R/dependencies.R (hunk 3 above)
Comment (kevinushey) [reply R/dependencies.R:395]: Yes, see https://github.com/rstudio/renv/pull/1603/files#diff-5d849f3782ddfdd6b0053b841b4be729f4fdaa33c9262caf6757aa185e5f5658R927-R929, which fills in the `count` variable, and https://github.com/rstudio/renv/pull/1603/files#diff-5d849f3782ddfdd6b0053b841b4be729f4fdaa33c9262caf6757aa185e5f5658R946-R948 where we use it.

**Comment by kevinushey** [comment]
I'm going to merge this and consider your comments as follow-up for https://github.com/rstudio/renv/issues/1607.

**Pull request merged**